Shuttle-back/src/main/java/com/shuttle/driver/DriverController.java

```diff
@@ -124,17 +143,23 @@ public ResponseEntity<WorkHours> changeWorkHours(@PathVariable(value = "working-
 
     }
 
+    @PermitAll
     @GetMapping("/api/driver/{id}/ride")
-    public ResponseEntity<ListDTO<RideDTO>> getRideHistory(@PathVariable(value = "id") Long id,
-                                                           @PathParam("page") int page, @PathParam("size") int size,
-                                                           @PathParam("from") String from, @PathParam("to") String to,
-                                                           @PathParam("to") String sort) {
-        ListDTO<RideDTO> rideDTOListDTO = new ListDTO<>();
-        rideDTOListDTO.setTotalCount(page);
-        List<RideDTO> rideDTOList = new ArrayList<>();
-        rideDTOList.add(new RideDTO());
-        rideDTOListDTO.setResults(rideDTOList);
-        return new ResponseEntity<>(rideDTOListDTO, HttpStatus.OK);
+    public ResponseEntity<ListDTO<Ride>> getRideHistory(@PathVariable(value = "id") Long id,
+                                                     @PathParam("page") int page, @PathParam("size") int size,
+                                                     @PathParam("from") String from, @PathParam("to") String to,
+                                                     @PathParam("to") String sort) {
+        Pageable pageParams =
```

Comment:
You can replace `@PathParam("page") int page, @PathParam("size") int size` in the parameter list of this method with just `Pageable pageParams` and Spring will take care of everything. It's less code and it works exactly the same.